pipeline/mutate/mutator_hydrator.go

```diff
@@ -78,21 +84,53 @@ type externalAPIConfig struct {
 }
 
 type MutatorHydratorConfig struct {
-	Api externalAPIConfig `json:"api"`
+	Api      externalAPIConfig `json:"api"`
+	CacheTtl string            `json:"cache_ttl"`
 }
 
 type mutatorHydratorDependencies interface {
 	x.RegistryLogger
 }
 
 func NewMutatorHydrator(c configuration.Provider, d mutatorHydratorDependencies) *MutatorHydrator {
-	return &MutatorHydrator{c: c, d: d, client: httpx.NewResilientClientLatencyToleranceSmall(nil)}
+	cache, _ := ristretto.NewCache(&ristretto.Config{
+		NumCounters: 10000,
+		MaxCost:     1 << 25,
+		BufferItems: 64,
+	})
+	return &MutatorHydrator{c: c, d: d, client: httpx.NewResilientClientLatencyToleranceSmall(nil), hydrateCache: cache}
 }
 
 func (a *MutatorHydrator) GetID() string {
 	return "hydrator"
 }
 
+func (a *MutatorHydrator) cacheKey(config *MutatorHydratorConfig, session *authn.AuthenticationSession) string {
+	return fmt.Sprintf("%x",
+		md5.Sum([]byte(fmt.Sprintf("%s|%s", config.Api.URL, session.Subject))),
```

Comment:
Since the string is not that long I think we can use it without md5 as it saves a couple of CPU rounds :)